fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanChargeWritePlatformServiceImpl.java

```diff
@@ -509,6 +509,10 @@ public CommandProcessingResult waiveLoanCharge(final Long loanId, final Long loa
                 existingTransactionIds, existingReversedTransactionIds, loanInstallmentNumber, scheduleGeneratorDTO, accruedCharge,
                 externalId);
 
+        if (loanCharge.isInstalmentFee()) {
```

Comment:
why do we need this? Waive loan charge functionality is not really relating to the transaction processing of reverse-replaying...